pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -40,17 +41,60 @@ type VolumeRestrictions struct {
 var _ framework.PreFilterPlugin = &VolumeRestrictions{}
 var _ framework.FilterPlugin = &VolumeRestrictions{}
 var _ framework.EnqueueExtensions = &VolumeRestrictions{}
-
-// Name is the name of the plugin used in the plugin registry and configurations.
-const Name = names.VolumeRestrictions
+var _ framework.StateData = &preFilterState{}
 
 const (
+	// Name is the name of the plugin used in the plugin registry and configurations.
+	Name = names.VolumeRestrictions
+	// preFilterStateKey is the key in CycleState to VolumeRestrictions pre-computed data for Filtering.
+	// Using the name of the plugin will likely help us avoid collisions with other plugins.
+	preFilterStateKey = "PreFilter" + Name
+
 	// ErrReasonDiskConflict is used for NoDiskConflict predicate error.
 	ErrReasonDiskConflict = "node(s) had no available disk"
 	// ErrReasonReadWriteOncePodConflict is used when a pod is found using the same PVC with the ReadWriteOncePod access mode.
 	ErrReasonReadWriteOncePodConflict = "node has pod using PersistentVolumeClaim with the same name and ReadWriteOncePod access mode"
 )
 
+// ErrStatusKeyNotFound is returned when the plugin PreFilter state cannot be read.
+var ErrStatusKeyNotFound = fmt.Errorf("cannot read %q from cycleState", preFilterStateKey)
```

Comment:
Nit: seems no need to export this error.

Reply:
This should be enough

```suggestion
var errStatusKeyNotFound = fmt.Errorf("cannot read %q from cycleState", preFilterStateKey)
```